dynamic_rest/client/client.py

```diff
@@ -0,0 +1,187 @@
+import json
+import requests
+from .exceptions import AuthenticationFailed, BadRequest, DoesNotExist
+from .resource import DRESTResource
+from dynamic_rest.conf import settings
+
+
+class DRESTClient(object):
+    """DREST Python client.
+
+    Exposes a DREST API to Python using a Django-esque interface.
+    Resources are available on the client through access-by-name.
+
+    Arguments:
+        host: hostname to a DREST API
+        version: version (defaults to no version)
+        client: HTTP client (defaults to requests.session)
+        scheme: defaults to https
+        authentication: if unset, authentication is disabled.
+            If set, provides credentials: {
+                usename: login username,
+                password: login password,
+                token: authorization token,
+                cookie: session cookie
+            }
+            Either username/password, token, or cookie should be provided.
+
+    Examples:
+
+    Getting a client:
+
+        client = DRESTClient('my.api.io', authentication={'token': 'secret'})
+
+    Working with a resource:
+
+        Users = client.Users
+
+    Getting a single resource:
+
+        Users.get('123')
+
+    Getting all resources (auto-paging):
+
+        Users.all()
+
+    Getting filtered resources:
+
+        Users.filter(name__icontains='john')
+        other_users = client.users.exclude(name__icontains='john')
+
+    Including / excluding fields:
+
+        users = Users.all()
+        .excluding('birthday')
+        .including('events.*')
+        .get('123')
+
+    Mapping by field:
+
+        users_by_id = Users.map()
+        users_by_name = Users.map('name')
+
+    Ordering results:
+
+        users = Users.order_by('-name')
```

Comment:
The DREST param by default is `sort[]`... would it be better to be consistent with that? This is more consistent with Django though.